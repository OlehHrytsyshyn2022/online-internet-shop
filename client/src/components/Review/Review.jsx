import React from "react";

import { ListItem, ListItemText, Typography } from "@mui/material";

import Rating from "@mui/material/Rating";

import DeleteForeverIcon from "@mui/icons-material/DeleteForever";

const Review = ({ name, date, review, rating, onDelete }) => {
  return (
    <>
      <ListItem
        alignItems="flex-start"
        sx={{ padding: 0, marginBottom: "50px", maxWidth: "auto" }}
      >
        <ListItemText
          primary={
            <React.Fragment>
              <Typography
                sx={{
                  fontSize: "20px",
                  fontWeight: 400,
                  lineHeight: "26px",
                  marginLeft: "-2px",
                }}
                component="span"
                variant="body1"
                color="text.primary"
              >
                {name}
              </Typography>
              <Typography
                sx={{ display: "inline", marginLeft: "10px" }}
                component="span"
                variant="body2"
                color="text.primary"
              >
                {date}
              </Typography>
            </React.Fragment>
          }
          secondary={
            <React.Fragment>
              <Rating
                name="read-only"
                value={rating}
                readOnly
                sx={{ marginTop: "7px", marginLeft: "-5px" }}
                precision={0.5}
              />
              <Typography
                component="span"
                variant="body1"
                color="text.primary"
                sx={{
                  display: "block",
                  fontSize: "16px",
                  fontWeight: 400,
                  lineHeight: "27px",
                  wordWrap: "break-word",
                }}
              >
                {review}
              </Typography>
            </React.Fragment>
          }
        />

        <DeleteForeverIcon
          fontSize="medium"
          sx={{ color: "#707070", marginRight: "5px" }}
          onClick={onDelete}
        />
      </ListItem>
    </>
  );
};

export default Review;
