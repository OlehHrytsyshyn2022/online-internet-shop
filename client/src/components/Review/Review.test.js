import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Review from "./Review";

describe("Review", () => {
  test("renders Review component", () => {
    const review = {
      name: "Dmytro",
      date: "2023-09-15",
      review: "Great product!",
      rating: 4.5,
      onDelete: jest.fn(),
    };

    const { container } = render(<Review {...review} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onClick handler when clicked", () => {
    const review = {
      name: "Dmytro",
      date: "2023-09-15",
      review: "Great product!",
      rating: 4.5,
      onDelete: jest.fn(),
    };

    render(<Review {...review} />);
    fireEvent.click(screen.getByTestId("DeleteForeverIcon"));
    expect(review.onDelete).toHaveBeenCalled();
  });
});
