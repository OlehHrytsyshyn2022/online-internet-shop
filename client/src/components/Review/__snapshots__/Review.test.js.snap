// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders Review component 1`] = `
<li
  class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart css-1jvt0gr-MuiListItem-root"
>
  <div
    class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
  >
    <span
      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 css-1pis36y-MuiTypography-root"
      >
        Dmytro
      </span>
      <span
        class="MuiTypography-root MuiTypography-body2 css-1m2gncc-MuiTypography-root"
      >
        2023-09-15
      </span>
    </span>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
    >
      <span
        aria-label="4.5 Stars"
        class="MuiRating-root MuiRating-sizeMedium Mui-readOnly MuiRating-readOnly css-179v0gk-MuiRating-root"
        role="img"
      >
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 0%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
        <span
          class="MuiRating-decimal css-34he1w-MuiRating-decimal"
        >
          <span
            style="width: 50%; overflow: hidden; position: absolute;"
          >
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                />
              </svg>
            </span>
          </span>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 css-ps9zsu-MuiTypography-root"
      >
        Great product!
      </span>
    </p>
  </div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10toome-MuiSvgIcon-root"
    data-testid="DeleteForeverIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
    />
  </svg>
</li>
`;
